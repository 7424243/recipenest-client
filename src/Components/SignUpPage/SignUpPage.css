input {
    margin-bottom: 10px;
}

input, label {
    display: block;
    width: 100%;
}

label {
    margin-bottom: 5px;
}

.buttons {
    margin-top: 40px;
    text-align: center;
}

.sign-up-form {
    margin: 0 auto;
    padding: 0 0 40px;
    max-width: 450px;
    text-align: left;
}