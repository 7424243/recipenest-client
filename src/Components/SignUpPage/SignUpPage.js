//To-Do:
//Add PropTypes
//Add Error Message(s)

import React, {Component} from 'react'
import { Link } from 'react-router-dom'
import config from '../../config'
import RecipenestContext from '../../RecipenestContext'
import ErrorBoundary from '../ErrorBoundary/ErrorBoundary'
import './SignUpPage.css'
import PropTypes from 'prop-types'

class SignUpPage extends Component {

    constructor(props) {
        super(props)
        this.state = {
            full_name: '',
            user_name: '',
            password: '',
            nickname: '',
            error: null
        }
    }


    static contextType = RecipenestContext

    handleSubmit = e => {
        e.preventDefault()
        fetch(`${config.API_ENDPOINT}/users`, {
            method: 'POST',
            headers: {
                'content-type': 'application/json',
            },
            body: JSON.stringify(this.state)
        })
            .then(res => {
                if(!res.ok) {
                    return res.json().then(e => Promise.reject(e))
                }
                return res.json()
            })
            .then(() => {
                this.context.onSignUpSuccess()
                this.props.history.push('/login')
            })
            .catch(err => {
                this.setState({error: err.error})
                console.error({err})
            })
    }

    handleAddFullname = e => {
        this.setState({full_name: e.target.value})
    }

    handleAddUsername = e => {
        this.setState({user_name: e.target.value})
    }

    handleAddPassword = e => {
        this.setState({password: e.target.value})
    }

    handleAddNickname = e => {
        this.setState({nickname: e.target.value})
    }


    render() {
        return (
            <div className='sign-up-container'>
                <h3>Sign Up!</h3>
                <ErrorBoundary>
                        <form 
                            className='sign-up-form'
                            onSubmit={this.handleSubmit}
                        >
                            <p className='required-field'>* required field</p>
                            <section className='fullname'>
                                <label htmlFor='fullname'>Full Name: * </label>
                                <input 
                                    type='text' 
                                    placeholder='Full Name'
                                    required 
                                    onChange={this.handleAddFullname}
                                />
                            </section>
                            <section className='url'>
                                <label htmlFor='username'>Username: * </label>
                                <input 
                                    type='text' 
                                    placeholder='username' 
                                    required
                                    onChange={this.handleAddUsername}
                                />
                                
                            </section>
                            <section className='password'>
                                <label htmlFor='password'>Password: * </label>
                                <p>Must be at least 8 characters long and contain at least 1 upper case, 1 lower case, 1 number, and 1 special character.</p>
                                <input 
                                    type='password' 
                                    placeholder='password' 
                                    required
                                    onChange={this.handleAddPassword}
                                />
                            </section>
                            <section className='nickname'>
                                <label htmlFor='nickname'>Nickname (optional): </label>
                                <input 
                                    type='text' 
                                    placeholder='optional nickname' 
                                    onChange={this.handleAddNickname}/>
                            </section>
                            <section className='buttons'>
                                <Link to='/'><button>Cancel</button></Link>
                                <button type='submit'>Submit</button>
                            </section>
                            {this.state.error && <p>{this.state.error.message}. Please try again.</p>}
                        </form>
                </ErrorBoundary>
                
            </div>
        )
    }
}

export default SignUpPage

SignUpPage.propTypes = {
    history: PropTypes.shape({
        push: PropTypes.func
    }).isRequired
}