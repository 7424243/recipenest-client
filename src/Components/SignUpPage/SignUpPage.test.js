import React from 'react'
import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'
import SignUpPage from './SignUpPage';

describe('SignUpPage Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><SignUpPage/></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })  

  it('renders the UI as expected', () => {
    const tree = renderer 
      .create(<BrowserRouter><SignUpPage/></BrowserRouter>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
