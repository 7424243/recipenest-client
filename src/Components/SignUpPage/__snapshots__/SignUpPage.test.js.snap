// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpPage Component renders the UI as expected 1`] = `
<div
  className="sign-up-container"
>
  <h3>
    Sign Up!
  </h3>
  <form
    className="sign-up-form"
    onSubmit={[Function]}
  >
    <section
      className="fullname"
    >
      <label
        htmlFor="fullname"
      >
        Full Name: 
      </label>
      <input
        onChange={[Function]}
        placeholder="Full Name"
        required={true}
        type="text"
      />
    </section>
    <section
      className="url"
    >
      <label
        htmlFor="username"
      >
        Username: 
      </label>
      <input
        onChange={[Function]}
        placeholder="username"
        required={true}
        type="text"
      />
    </section>
    <section
      className="password"
    >
      <label
        htmlFor="password"
      >
        Password: 
      </label>
      <input
        onChange={[Function]}
        placeholder="password"
        required={true}
        type="password"
      />
    </section>
    <section
      className="nickname"
    >
      <label
        htmlFor="nickname"
      >
        Nickname (optional): 
      </label>
      <input
        onChange={[Function]}
        placeholder="optional nickname"
        type="text"
      />
    </section>
    <section
      className="buttons"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <button>
          Cancel
        </button>
      </a>
      <button
        type="submit"
      >
        Submit
      </button>
    </section>
  </form>
</div>
`;
